packages/volto/src/hooks/user/useFetchUser.js

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import jwtDecode from 'jwt-decode';
+import { getUser } from '@plone/volto/actions';
+
+const useFetchUser = () => {
+  const user = useSelector((state) => state.users?.user);
+  const userId = useSelector((state) =>
+    state.userSession.token ? jwtDecode(state.userSession.token).sub : '',
+  );
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (!user?.id) {
+      dispatch(getUser(userId));
+    }
+  }, [dispatch, userId, user]);
```

Comment:
If there are multiple components which use this hook, they might dispatch multiple copies of the getUser action at the same time.

Maybe there's a way to instead make sure that the user data is always fetched immediately after the user logs in.

Reply:
Done